### DISCUSSION TITLE
Am I using npm-check-updates with Yarn 4 correctly?

### DISCUSSION THREAD

**Discussion by Jayman2000** [raineorshine/npm-check-updates#1464] [Q&A]
Hi,

I’m really new to Node.js, and I’m struggling to figure out how to do updates for Node.js-related things.

The project that I’m currently working on has this in its `package.json` file:

```json
  "packageManager": "yarn@4.4.0"
```

I would like to be able to run a command in order to bump the version in the `"packageManager"` field. At the moment, the latest version of Yarn 4 is [4.5.0](https://github.com/yarnpkg/berry/releases/tag/%40yarnpkg%2Fcli%2F4.5.0), but when I run `npx npm-check-updates --dep packageManager`, it says that everything is up to date. That being said, if I change my `package.json` file so that it says this:

```json
  "packageManager": "yarn@1.22.0"
```

Then, `npx npm-check-updates --dep packageManager` will tell me that I can upgrade to Yarn [1.22.22](https://github.com/yarnpkg/yarn/releases/tag/v1.22.22) which is the latest stable version of Yarn Classic.

It seems like there’s one of two things going on here:

1. The `"packageManager"` field in my `package.json` file isn’t correct. I need to do something else in order to use Yarn 4.
2. There’s a bug with npm-check-updates.

Which one of those possibilities is happening here?

**Comment by raineorshine** [comment]
Hi, thanks for your question.

There's a third option! Yarn 4 is no longer published to the npm registry. It uses Corepack and its own hosting service. npm-check-updates, while compatible with other package managers, only works with the npm registry or private registries with the same API.

The latest version of `yarn` on the npm registry is v1.22.22:

```sh
$ npm view yarn version
1.22.22
```

**Reply by Jayman2000** [reply]
OK, so I guess that npm-check-updates isn’t the right tool to use here since Yarn 4 isn’t published to a npm-style registry. Is there another tool that does the same thing as `npx npm-check-updates --dep packageManager -u` but works with Yarn 4?

**Reply by raineorshine** [reply]
Yes, exactly. 

I'm afraid I don't know of other tools for yarn, but I haven't really looked.
